File: vaidcord-go/channel.go
package vaidcord

type Channel struct {
	ID                         string           `json:"id"`
	Type                       int              `json:"type"`
	GuildID                    string           `json:"guild_id,omitempty"`
	Name                       string           `json:"name,omitempty"`
	Topic                      string           `json:"topic,omitempty"`
	Position                   int              `json:"position,omitempty"`
	NSFW                       bool             `json:"nsfw,omitempty"`
	ParentID                   string           `json:"parent_id,omitempty"`
	LastMessageID              string           `json:"last_message_id,omitempty"`
	RateLimitPerUser           int              `json:"rate_limit_per_user,omitempty"`
	PermissionOverwrites       []map[string]any `json:"permission_overwrites,omitempty"`
	DefaultAutoArchiveDuration int              `json:"default_auto_archive_duration,omitempty"`
}

File: vaidcord-rust/src/models/channel.rs
use serde::Deserialize;
use serde_json::Value;

#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
pub struct Channel {
    pub id: String,
    #[serde(rename = "type")]
    pub kind: u8,
    #[serde(default)]
    pub guild_id: Option<String>,
    #[serde(default)]
    pub name: Option<String>,
    #[serde(default)]
    pub topic: Option<String>,
    #[serde(default)]
    pub position: Option<i64>,
    #[serde(default)]
    pub nsfw: Option<bool>,
    #[serde(default)]
    pub parent_id: Option<String>,
    #[serde(default)]
    pub last_message_id: Option<String>,
    #[serde(default)]
    pub rate_limit_per_user: Option<u64>,
    #[serde(default)]
    pub permission_overwrites: Vec<Value>,
    #[serde(default)]
    pub default_auto_archive_duration: Option<u64>,
}
